src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Hover/DefaultRazorHoverInfoService.cs

```diff
@@ -162,18 +157,18 @@ public override HoverModel GetHoverInfo(RazorCodeDocument codeDocument, SourceLo
                     }
 
                     var attributeName = attribute.GetContent();
-                    var range = attribute.GetRange(codeDocument.Source);
+                    var range = attribute.GetVSRange(codeDocument.Source);
 
                     // Include the @ in the range
                     switch (attribute.Parent.Kind)
                     {
                         case SyntaxKind.MarkupTagHelperDirectiveAttribute:
-                            var directiveAttribute = attribute.Parent as MarkupTagHelperDirectiveAttributeSyntax;
+                            var directiveAttribute = (attribute.Parent as MarkupTagHelperDirectiveAttributeSyntax)!;
                             range.Start.Character -= directiveAttribute.Transition.FullWidth;
                             attributeName = "@" + attributeName;
                             break;
                         case SyntaxKind.MarkupMinimizedTagHelperDirectiveAttribute:
-                            var minimizedAttribute = containingTagNameToken.Parent as MarkupMinimizedTagHelperDirectiveAttributeSyntax;
+                            var minimizedAttribute = (containingTagNameToken.Parent as MarkupMinimizedTagHelperDirectiveAttributeSyntax)!;
```

Comment:
For future, a cast is better for these situations so if anything goes wrong, you get "cannot cast blah to blah" on this line, rather than "null reference" on some other line later.